src/main/java/sx/blah/discord/modules/ModuleLoader.java

```diff
@@ -356,4 +271,74 @@ private static File findFileForClass(List<File> files, String clazz) {
 	public static void addModuleClass(Class<? extends IModule> clazz) {
 		modules.add(clazz);
 	}
+
+	/**
+	 * Gets the modules loaded in this ModuleLoader instance.
+	 *
+	 * @return The list of loaded modules.
+	 */
+	public List<IModule> getLoadedModules() {
+		return loadedModules;
+	}
+
+	/**
+	 * Manually loads a module.
+	 *
+	 * @param module The module to load.
+	 * @return True if the module was successfully loaded, false if otherwise. Note: successful load != successfully enabled
+	 */
+	public boolean loadModule(IModule module) {
+		Class<? extends IModule> clazz = module.getClass(); if (clazz.isAnnotationPresent(Requires.class)) {
+			Requires annotation = clazz.getAnnotation(Requires.class);
+			if (!hasDependency(loadedModules, annotation.value())) {
+				return false;
+			}
+		} boolean enabled = module.enable(client); if (enabled) {
+			client.getDispatcher().registerListener(module);
+			if (!loadedModules.contains(module)) loadedModules.add(module);
+
+			client.getDispatcher().dispatch(new ModuleEnabledEvent(module));
+		}
+
+		return true;
+	}
+
+	/**
+	 * Manually unloads a module.
+	 *
+	 * @param module The module to unload.
+	 */
+	public void unloadModule(IModule module) {
+		loadedModules.remove(module); module.disable(); client.getDispatcher().unregisterListener(module);
+
+		loadedModules.removeIf(mod -> {
+			Class<? extends IModule> clazz = module.getClass(); if (clazz.isAnnotationPresent(Requires.class)) {
+				Requires annotation = clazz.getAnnotation(Requires.class);
+				if (annotation.value().equals(module.getClass().getName())) {
+					unloadModule(mod); return true;
+				}
+			} return false;
+		});
+
+		client.getDispatcher().dispatch(new ModuleDisabledEvent(module));
+	}
+
+	private boolean hasDependency(List<IModule> modules, String className) {
+		for (IModule module : modules)
+			if (module.getClass().getName().equals(className)) return true; return false;
+	}
+
+	private boolean canModuleLoad(IModule module) {
+		String[] versions; String[] discord4jVersion; try {
+			versions = module.getMinimumDiscord4JVersion().toLowerCase().replace("-snapshot", "").split("\\.");
+			discord4jVersion = Discord4J.VERSION.toLowerCase().replace("-snapshot", "").split("\\.");
+
+			for (int i = 0; i < Math.min(versions.length, 2); i++) { //We only care about major.minor, the revision change should not be big enough to care about
+				if (Integer.parseInt(versions[i]) > Integer.parseInt(discord4jVersion[i])) return false;
```

Comment:
Inconsistency with single-line if statements being on the second line. (Not a big deal)